core/invoker/src/main/scala/whisk/core/containerpool/ContainerProxy.scala

```diff
@@ -374,6 +374,10 @@ object ContainerProxy {
 
     // Needs to be thread-safe as it's used by multiple proxies concurrently.
     private val containerCount = new Counter
+    private var instance: Int = 0
+
+    def instanceId(instanceId: Int) =
+        instance = instanceId
```

Comment:
Alternative idea: Pass the instance as an environment variable to the invoker as well and read it here statically?